config/src/test/groovy/org/springframework/security/config/annotation/provisioning/UserDetailsManagerConfigurerTests.groovy

```diff
@@ -30,10 +32,8 @@ import spock.lang.Specification
 class UserDetailsManagerConfigurerTests extends Specification {
 
 	def "all attributes supported"() {
-		setup:
-			InMemoryUserDetailsManager userDetailsManager = new InMemoryUserDetailsManager([])
 		when:
-			UserDetails userDetails = new UserDetailsManagerConfigurer<InMemoryUserDetailsManager,UserDetailsManagerConfigurer<InMemoryUserDetailsManager>>(userDetailsManager)
```

Comment:
These tests are suppose to be validating `UserDetailsManagerConfigurer` not `InMemoryUserDetailsManagerConfigurer`. Is there a reason you switched this?

Reply:
In fact I did a mistake here. But rather than parameterizing `UserDetailsManagerConfigurer` with `InMemoryUserDetailsManager` I think it should be better to use `AuthenticationManagerBuilder` as follows:

```java
UserDetails userDetails = new UserDetailsManagerConfigurer<AuthenticationManagerBuilder,
    UserDetailsManagerConfigurer<InMemoryUserDetailsManager>>(userDetailsManager)
```

That's because as the first generic parameter type for `UserDetailsManagerConfigurer` which is `B` extends `ProviderManagerBuilder`,  I realized that it should make sense to use `AuthenticationManagerBuilder`.

If I keep using `InMemoryUserDetailsManager` for the first generic parameter type, I got the following error on STS:
```
Groovy:The type InMemoryUserDetailsManager is not a valid substitute 
for the bounded parameter 
<B extends org.springframework.security.config.annotation.authentication.ProviderManagerBuilder<B>>
```

I made the changes to use `AuthenticationManagerBuilder` as proposed but did not pushed yet just to preserve our conversation here. If you think it makes sense to be applied I can submit the changes.